### DISCUSSION TITLE
'rack.attack.match_data' not available when logging notifications?

### DISCUSSION THREAD

**Discussion by gregblass** [kickstarter/rack-attack#527] [Questions (Q&A)]
I'm trying to save the first time someone gets banned to the database for logging. I only want to do the first time, since subsequent times could overload our database if a spammer hits our server.

I found a couple threads recommending to check the request's `env['rack.attack.match_data']`.

However, in 6.2.2, that isn't coming through for me:

```
ActiveSupport::Notifications.subscribe("blocklist.rack_attack") do |name, start, finish, request_id, payload|
  req = payload[:request]
  puts req.env['rack.attack.match_data'] # => nil
end
```

Was that removed? Or am I missing something here? Or is there another preferred way now of finding out if this was the request that actually caused the ban?

**Comment by gregblass** [comment]
Ahhh - it looks like this is only available for the throttle.rack_attack event.

How would one go about checking the request count for Fail2Ban? (blocklist.rack_attack)

**Comment by buithehien1991** [comment]
Could you tell me what event for blocklist? (blocklist.rack_attack right ?)

Thanks
